### ISSUE TITLE
[no-shadow] False positive with const enum and module augmentation

### ISSUE THREAD

**Issue by sveyret** [typescript-eslint/typescript-eslint#2484]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "no-shadow": ["error", { "hoist": "all" }]
  }
}
```

```TS
export const enum MyEnum {
  // error  'MyEnum' is already declared in the upper scope  no-shadow
  Value1,
  Value2,
  Value3,
}

declare module '@material-ui/core/styles/createMuiTheme' {
  interface Theme {
    // error  'Theme' is already declared in the upper scope  no-shadow
    windowSize: number
  }
}
```

**Expected Result**

There should be no eslint error, because nothing is shadowed.

**Actual Result**

Eslint indicate an error for both the const enum and the re-defined interface.

**Additional Info**

This is a regression, my code was compiling without complaining before I upgraded to latest versions.
I write this issue here because it is related to Typescript and may also be related to the correction of #325.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.0.1` |
| `@typescript-eslint/parser`        | `4.0.1` |
| `TypeScript`                       | `4.0.2` |
| `ESLint`                           | `7.8.1` |
| `node`                             | `14.4.0` |

**Comment by tadhgmister** [comment]
are you using the base eslint rule or ours? the config should look something like

```jsonc
{
  // note you must disable the base rule as it can report incorrect errors
  "no-shadow": "off",
  "@typescript-eslint/no-shadow": ["error"]
}
```
as mentioned [in the docs](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-shadow.md)

**Comment by bradzacher** [comment]
@tadhgmister (thanks!) is correct - you need to update your config 

- [This is a listed action item for users in the v4.0.0 release notes](https://github.com/typescript-eslint/typescript-eslint/releases/tag/v4.0.0.)
- [In case you missed the release note, this is listed in the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md#i-am-using-a-rule-from-eslint-core-and-it-doesnt-work-correctly-with-typescript-code)
- [Our extension rule's docs lists the config you should use](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-shadow.md)

Finally, please use the search next time.
Duplicate of #2466 and #2483
